torch/_logging/_registrations.py

```diff
@@ -132,3 +132,4 @@
 )
 
 register_artifact("custom_format_test_artifact", "Testing only", log_format="")
+register_artifact("not_implemented", "Logs dispatches not implemented by Fake Tensor")
```

Comment:
As per offline discussion with @atalman, this may not be needed. It seems there is a race condition and in some internal tests, `torch._guards.detect_fake_mode(None)` is executed before `register_artifact()` is executed.

He was looking into moving the `register_artifact`call to a function instead of having it hanging as a global

Reply:
I am seeing us that  https://github.com/pytorch/pytorch/blob/main/torch/_subclasses/fake_tensor.py?plain=1 is executed before registering is happening. Moving the ``not_implemented_log = torch._logging.getArtifactLogger(__name__, "not_implemented")`` to a function seems to resolve this

Reply:
`not_implemented` is already registered at https://github.com/pytorch/pytorch/blob/main/torch/_logging/_registrations.py#L101-L104

Reply:
@thiagocrepaldi yes indeed, let me see if I can provide small repro fo this. Maybe test is doing something wrong ?